src/service/service.ts

```diff
@@ -386,7 +386,11 @@ export class Discv5 extends (EventEmitter as { new (): Discv5EventEmitter }) {
   /**
    * Send TALKRESP message to requesting node
    */
-  public async sendTalkResp(remote: ENR | Multiaddr, requestId: RequestId, payload: Uint8Array): Promise<void> {
+  public async sendTalkResp(
+    remote: ENR | Multiaddr | INodeAddress,
```

Comment:
Can you change this to only allow `nodeAddr: INodeAddress`?

And revert the changes to `createNodeContact`?

Reply:
To make sure I'm tracking, you want `remote : INodeAddress`?  If so, don't we need to update the function signature of `sendRpcResponse`  as well since it wants a `NodeContact` and not just the `INodeAddress`.  The challenge this gives for Portal Network (unless I'm missing something) is that we may not have access to a node's public key, just an invalid ENR that contains the nodeId.  We can use this `nodeId` plus the observed address to still specify which session to use but there's no clear way to get the full Multiaddr with an inlined public key from that since as far as I understand you can't go from a nodeId back to a public key.

Reply:
Disregard.  I was looking at the wrong method.  This feedback should now be addressed.